src/cluster/placement/selector/mirrored.go

```diff
@@ -226,34 +231,75 @@ func (f *mirroredSelector) SelectReplaceInstances(
 			continue
 		}
 
+		for _, group := range groups {
+			if len(group) != 2 {
+				return nil, fmt.Errorf(
+					"unexpected length of instance group for replacement: %d",
+					len(group),
+				)
+			}
+			replacementGroups = append(
+				replacementGroups,
+				mirroredReplacementGroup{
+					Leaving:     group[0],
```

Comment:
how do you know [0] is leaving and [1] is the replacement?

Reply:
Good question--that was the previous convention; presumably it remains true. I think the unittests will catch it; I'll double check.

Reply:
Yeah basically this is a side effect of the way `groupInstancesByHostPort`  is implemented. I added a unittest that tries to assert that behavior, and also a check in the code path itself; hopefully sufficient.

Reply:
as discussed offline: check both [0]/[1] to see which is leaving, and which isn't

Reply:
👍 new code fixes this I think.